### DISCUSSION TITLE
Filter based on parameter type

### DISCUSSION THREAD

**Discussion by akam-it** [fastapi/fastapi#6438] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@router.get('/{host_id}', response_model=Host)
def get_siphost(host_id: Union[int, str], service: HostsService = Depends()):
    return service.get(host_id)

-----

    def _get(self, host_id: Union[int, str]) -> tables.Host:
        siphost = (
            self.session
            .query(tables.Host)
            .filter_by(id=host_id)   <<<<< is it possible to do dynamic filter based on type int or str
            .first()
        )
        if not host:
            raise HTTPException(status.HTTP_404_NOT_FOUND)
        return host
```

### Description

Hello, sorry for a noobs question, didn't find it in documentation :(

I have a simple route /host with a simple model - id, name (all unique)

is it possible to get a host by id or name on a single path? I mean if /hosts/100 - will be returned host with id 100, if /hosts/localhost - will be returned host with hostname localhost.

### Operating System

Linux

### FastAPI Version

0.70.0

### Python Version

Python 3.10

**Comment by panla** [comment, marked as answer]
do you mean ?

```python
from sqlalchemy.sql import or_

.filter_by(or_(id=host_id, host=host_id))
```

**Comment by akam-it** [comment]
thank you very much @panla!

in my case this worked like this:
.filter(or_(tables.Host.id==host_id, tables.Host.name==host_id))
